plaso/parsers/fseventsd.py

```diff
@@ -159,8 +153,23 @@ def ParseFileObject(self, parser_mediator, file_object, **kwargs):
     Raises:
       UnableToParseFile: when the header cannot be parsed.
     """
-    page_header = self._ParseDLSPageHeader(file_object)
+    page_header_map = self._GetDataTypeMap('dls_page_header')
+    file_offset = 0
+    file_size = file_object.get_size()
+    try:
+      page_header, header_size = self._ReadStructureFromFileObject(
+          file_object, file_offset, page_header_map)
+    except (ValueError, errors.ParseError) as exception:
+      raise errors.UnableToParseFile(
+          'Unable to parse page header with error: {0!s}'.format(
+              exception))
+
+    if page_header.signature not in self._DLS_SIGNATURES:
+      raise errors.UnableToParseFile('Invalid file signature')
+
+    file_offset += header_size
```

Comment:
Remove this if you apply previous suggestion

Reply:
Ack